tests/utils/stl/test/tests.py

```diff
@@ -7,215 +7,199 @@
-        return ':'.join((self.path_in_suite[-2], self.env_num))
+    def getTempPaths(self):
+        tmpDir = self.getExecDir()
+        tmpBase = os.path.join(tmpDir, self.path_in_suite[-2])
+        return tmpDir, os.path.normpath(tmpBase)
 
-    def getXMLOutputClassName(self):
-        safe_test_path = [x.replace(".", "_") for x in self.path_in_suite[:-1]]
-        safe_suite_name = self.suite.name.replace(".", "-")
+    def _configureExpectedResult(self, suite, litConfig):
+        testName = self.getTestName()
+        self.expectedResult = None
 
-        if safe_test_path:
-            return safe_suite_name + "." + "/".join(safe_test_path)
+        if testName in litConfig.expected_results.get(self.config.name, dict()):
+            self.expectedResult = litConfig.expected_results[self.config.name][testName]
         else:
-            return safe_suite_name + "." + safe_suite_name
-
-    def _configure_expected_result(self, suite, path_in_suite, lit_config,
-                                   test_config, env_num):
-        test_name = self.getTestName()
-        self.expected_result = None
-
-        current_prefix = ""
-        for prefix, result in \
-                chain(test_config.expected_results.items(),
-                      lit_config.expected_results.get(test_config.name,
-                                                      dict()).items()):
-            if test_name == prefix:
-                self.expected_result = result
-                break
-            elif test_name.startswith(prefix) and \
-                    len(prefix) > len(current_prefix):
-                current_prefix = prefix
-                self.expected_result = result
-
-        if test_name in test_config.expected_results:
-            self.expected_result = test_config.expected_results[test_name]
-        elif test_name in lit_config.expected_results:
-            self.expected_result = lit_config.expected_results[test_name]
-
-        if self.expected_result is not None:
-            if self.expected_result == SKIPPED:
-                self.skipped = True
-            elif self.expected_result.isFailure:
+          currentPrefix = ""
+          for prefix, result in litConfig.expected_results.get(self.config.name, dict()).items():
+              if testName == prefix:
+                  self.expectedResult = result
+                  break
+              elif testName.startswith(prefix) and len(prefix) > len(currentPrefix):
+                  currentPrefix = prefix
+                  self.expectedResult = result
+
+        if self.expectedResult is not None:
+            if self.expectedResult == SKIPPED:
+                self.result = Result(SKIPPED, 'This test was explicitly marked as skipped')
+            elif self.expectedResult.isFailure:
                 self.xfails = ['*']
+        elif self.config.unsupported:
+            self.result = Result(lit.Test.UNSUPPORTED, 'This test was marked as unsupported by a lit.cfg')

```

Comment:
It looks like this line should be:
```suggestion
            self.result = Result(UNSUPPORTED, 'This test was marked as unsupported by a lit.cfg')
```
but I'm only 80% sure.

Reply:
It can go either way.